### PR TITLE
Fix OSD preview

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #3100

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by nerdCopter** [comment]
hmm, well it does indeed fix the issue, but if the window width is less than ~1436, column 2 does not shrink enough and/or column 3 gets moved.

approx width 1436:
image

approx width 1432:
image

old behavior at approx width 1061:
image

File: src/css/tabs/osd.less
Comment (McGiverGim) [inline src/css/tabs/osd.less:353]: The problem here is the "size" of the char that contains the symbol. It is 12x18 and we don't reduce it because it almost disappears because the reduction algorithm done by the browser works very bad with character images.
Until now, PAL and NTSC have 30 columns, so 30 x 12 = 360 px (the CSS rule was wrong by 10 pixels). Now, in HD it contains 53 columns, so the minimum must be 53 x 12 = 636.
This will make difficult to fit all without flexing, so the better approach is to give different sizes for both systems, at least PAL and NTSC will fit better. We have the code to add a class, and use different backgronds depending on the video system before this PR https://github.com/betaflight/betaflight-configurator/pull/3071 that removed it because it was not needed anymore.
So, two options:
- Solution 1: simply use 636 as maximum
- Solution 2: add the code removed in the PR to add a class based on the video system, and use 360 by default and 636 for HD.

**Review by McGiverGim — CHANGES_REQUESTED** [review]

File: src/css/tabs/osd.less
Comment (haslinghuis) [reply src/css/tabs/osd.less]: Thanks. Using solution 1 now. Going to check solution 2.

**Comment by nerdCopter** [comment]
last commit `9f5e0e82` reverts to same behavior as bug and also limits minimal usable width.
this is approx 1332px at which anything smaller, then column 3 moves down.

image

**Comment by haslinghuis** [comment]
Reverted for now.

**Comment by howels** [comment]
Could we arrange the Elements and Active OSD/Video Format columns below Preview on screens <1432?

Text moves outside the Preview frame currently:
image

**Comment by howels** [comment]
Looks to be working well, resizes the preview area and then moves columns underneath.

https://user-images.githubusercontent.com/7933842/205521257-64dde951-34eb-4444-8e38-7b249dc35247.mp4

**Review by ctzsnooze — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Comment by howels** [comment]
Android OSD preview isn't completely broken but the contents need to be aligned left and scaled.

https://user-images.githubusercontent.com/7933842/205947586-619b612e-b979-4b7c-8686-1adf6f29d928.mp4

**Comment by haslinghuis** [comment]
On Samsung Galaxy S7 (SM-935F) with resolution 1440 x 2560 pixels the preview fills the screen (aligned left),
Only you can scroll to the right where there is white space,

What is the screen resolution of your device?

**Comment by McGiverGim** [comment]
In my case, pixel 4 with 2280 × 1080.

**Comment by howels** [comment]
> On Samsung Galaxy S7 (SM-935F) with resolution 1440 x 2560 pixels the preview fills the screen (aligned left), Only you can scroll to the right where there is white space,
> 
> What is the screen resolution of your device?

Sony Xperia 1 III, resolution 1644 x 3840. Guess it's down to the screen scaling. I am also running HD OSD.

**Comment by howels** [comment]
Could we scale the preview window on mobile? I know on desktop we want to avoid this but on mobile the touch targets are already challenging anyway. Losing fidelity in the fonts isn't a big deal on very small screens.

**Pull request merged**
